From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martin Finkel <me@martinfinkel.com>
Date: Wed, 5 Aug 2026 17:20:00 +0300
Subject: [PATCH] iconv: support UCRT HandleInformation declarations

Newer llvm-mingw UCRT headers declare GetHandleInformation for UWP but do
not expose HANDLE_FLAG_INHERIT. Keep VLC's fallback implementation for the
older UWP toolchain and only supply the missing flag for UCRT builds.
---
 ...call-GetHandleInformation-in-Winstore-apps.patch | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/contrib/src/iconv/0001-do-not-call-GetHandleInformation-in-Winstore-apps.patch b/contrib/src/iconv/0001-do-not-call-GetHandleInformation-in-Winstore-apps.patch
--- a/contrib/src/iconv/0001-do-not-call-GetHandleInformation-in-Winstore-apps.patch
+++ b/contrib/src/iconv/0001-do-not-call-GetHandleInformation-in-Winstore-apps.patch
@@ -9,14 +9,14 @@ is not allowed either [2].
 [1] https://docs.microsoft.com/en-us/windows/win32/api/handleapi/nf-handleapi-gethandleinformation
 [2] https://docs.microsoft.com/en-us/cpp/cppcx/crt-functions-not-supported-in-universal-windows-platform-apps
 ---
- lib/fcntl.c | 13 +++++++++++++
- 1 file changed, 13 insertions(+)
+ lib/fcntl.c | 15 +++++++++++++++
+ 1 file changed, 15 insertions(+)
 
 diff --git a/srclib/fcntl.c b/srclib/fcntl.c
 index e220800845..c95322b2d4 100644
 --- a/srclib/fcntl.c
 +++ b/srclib/fcntl.c
-@@ -45,6 +45,19 @@
+@@ -45,6 +45,21 @@
  #  include <io.h>
  # endif
  
@@ -25,11 +25,13 @@ index e220800845..c95322b2d4 100644
 +   are also not available, so we just return 0.
 +   <https://docs.microsoft.com/en-us/windows/win32/api/handleapi/nf-handleapi-gethandleinformation>
 +   <https://learn.microsoft.com/en-us/uwp/win32-and-com/win32-apis> */
++#  ifndef _UCRT
 +static inline BOOL GetHandleInformation(HANDLE h, DWORD *pf)
 +{
 +  *pf = 0;
 +  return TRUE;
 +}
++#  endif
 +#  define HANDLE_FLAG_INHERIT          (1)
 +# endif /*  WINAPI_PARTITION_DESKTOP */
 +
-- 
2.50.1.windows.1

